packages/teams-js/test/public/settings.spec.ts

```diff
@@ -138,120 +209,154 @@ describe('settings', () => {
-      handlerCalled = true;
+      const message = utils.findMessageByFunc('settings.remove.success');
+      expect(message).not.toBeNull();
+      expect(message.args.length).toBe(0);
     });
 
-    utils.sendMessage('settings.remove');
+    it('settings.registerOnRemoveHandler should successfully notify success from the registered remove handler', async () => {
+      await utils.initializeWithContext(FrameContexts.remove);
 
-    expect(handlerCalled).toBe(true);
-    const message = utils.findMessageByFunc('settings.remove.success');
-    expect(message).not.toBeNull();
-    expect(message.args.length).toBe(0);
-  });
+      let handlerCalled = false;
+      settings.registerOnRemoveHandler(removeEvent => {
+        removeEvent.notifySuccess();
+        handlerCalled = true;
+      });
 
-  it('should successfully notify failure from the registered remove handler', async () => {
-    await utils.initializeWithContext('remove');
+      utils.sendMessage('settings.remove');
 
-    let handlerCalled = false;
-    settings.registerOnRemoveHandler(removeEvent => {
-      removeEvent.notifyFailure('someReason');
-      handlerCalled = true;
+      expect(handlerCalled).toBe(true);
+      const message = utils.findMessageByFunc('settings.remove.success');
+      expect(message).not.toBeNull();
+      expect(message.args.length).toBe(0);
     });
 
-    utils.sendMessage('settings.remove');
+    it('settings.registerOnRemoveHandler should successfully notify failure from the registered remove handler', async () => {
+      await utils.initializeWithContext(FrameContexts.remove);
 
-    expect(handlerCalled).toBe(true);
-    const message = utils.findMessageByFunc('settings.remove.failure');
-    expect(message).not.toBeNull();
-    expect(message.args.length).toBe(1);
-    expect(message.args[0]).toBe('someReason');
-  });
+      let handlerCalled = false;
+      settings.registerOnRemoveHandler(removeEvent => {
+        removeEvent.notifyFailure('someReason');
+        handlerCalled = true;
+      });
 
-  it('should not allow multiple notifies from the registered save handler', async () => {
-    await utils.initializeWithContext('settings');
+      utils.sendMessage('settings.remove');
 
-    let handlerCalled = false;
-    settings.registerOnSaveHandler(saveEvent => {
-      saveEvent.notifySuccess();
-      expect(() => saveEvent.notifySuccess()).toThrowError('The SaveEvent may only notify success or failure once.');
-      expect(() => saveEvent.notifyFailure()).toThrowError('The SaveEvent may only notify success or failure once.');
-      handlerCalled = true;
+      expect(handlerCalled).toBe(true);
+      const message = utils.findMessageByFunc('settings.remove.failure');
+      expect(message).not.toBeNull();
+      expect(message.args.length).toBe(1);
+      expect(message.args[0]).toBe('someReason');
     });
+  });
 
+  it('should successfully notify success on save when there is no registered handler', async () => {
```

Comment:
I don't think this test case belongs with the removeHandler test cases. I think it's better grouped at the very beginning in the describe("settings") test group.

Reply:
Grouped to the top for the settings group.